pwiz_tools/Shared/CommonMsData/RemoteApi/WatersConnect/WatersConnectSession.cs

```diff
@@ -95,12 +149,13 @@ private ImmutableList<WatersConnectFolderObject> GetFolders(Uri requestUri)
             return ImmutableList.ValueOf(EnumerateChildFolderHierarchy(foldersValue.First() as JObject, null));
         }
 
-        private ImmutableList<WatersConnectFolderObject> GetInjections(Uri requestUri)
+        protected ImmutableList<WatersConnectFolderObject> GetInjections(Uri requestUri)
         {
-            var httpClient = WatersConnectAccount.GetAuthenticatedHttpClient();
-            var response = httpClient.GetAsync(requestUri).Result;
-            response.EnsureSuccessStatusCode();
+            var response = _httpClient.GetAsync(requestUri).Result;
+            EnsureSuccess(response);
             string responseBody = response.Content.ReadAsStringAsync().Result;
+            if (string.IsNullOrEmpty(responseBody))
+                return ImmutableList<WatersConnectFolderObject>.EMPTY;
```

Comment:
This seems like an error rather than a correct "empty folder" result. What prompted this change?